packages/components/tree-select/src/tree.ts

```diff
@@ -113,14 +121,11 @@ export const useTree = (
     onNodeClick: (data, node, e) => {
       attrs.onNodeClick?.(data, node, e)
 
-      if (
-        (props.checkStrictly
-          ? props.showCheckbox
-            ? props.checkOnClickNode
-            : props.checkStrictly
-          : props.checkStrictly) ||
-        node.isLeaf
-      ) {
+      // `onCheck` is trigger when `checkOnClickNode`
+      if (props.showCheckbox && props.checkOnClickNode) return
```

Comment:
Can I understand that the judgment logic of this part has been changed?

Reply:
No, checks are blocked from triggering twice here. In order to solve the problem of checking anomalies with only [show checkbox], see the new unit test. Their final result is the same, just as your previous unit tests have not been changed.

Reply:
All right